com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -451,6 +451,33 @@ internal NetworkTransformState GetLastSentState()
             return m_LastSentState;
         }
 
+        /// <summary>
+        /// This is invoked when a new client joins (server and client sides)
+        /// Server Side: Serializes as if we were teleporting (everything is sent via NetworkTransformState)
+        /// Client Side: Adds the interpolated state which applies the NetworkTransformState as well
+        /// </summary>
+        protected override void OnSynchronize<T>(ref BufferSerializer<T> serializer)
+        {
+            // We don't need to synchronize NetworkTransforms that are on the same
+            // GameObject as the NetworkObject.
```

Comment:
Wait, now I'm confused. What is `OnSynchronize`, and why does it involve multiple objects? Why is it not just "This NetworkTransform is on one object, we synchronize the position of this object" ?

Reply:
Marking this resolved after our 1:1 huddle.